### ISSUE TITLE
Additional decimal places in sub-obs lat & long in JPL Horizons breaks astroquery.jplhorizons parser

### ISSUE THREAD

**Issue by bholler** [astropy/astroquery#1647]
**Summary**: Additional decimal places were added to the sub-observer latitude and longitude for JPL Horizons ephemerides, resulting in errors when using the jplhorizons routine.

**Description**: I received the following email from the ssd-announce listserv, which may now be sending emails for updates that typically make it into the JPL Horizons systems news (however, at the time of submitting this ticket, the system news has yet to be updated):

```
Due to user request, the number of output digits for Horizons observer table
ephemeris quantities #14 and #15 (sub-observer and sub-solar longitude and
latitude) will be increased 4 decimal places each.

This is expected to go live on 2020-Feb-12 (Wednesday) and could be relevant
to users expecting fixed-format column-based data positions (or lengths) for
those two quantities.

Reason for change:

  The current 2 decimal place output can lead to rasterization and
  discontinuities in derived functions and plots based on the output.

  Increasing output to 6 decimals permits numerically smooth results,
  though the additional digits are often not physically meaningful due to
  most bodies having fairly low precision IAU rotation models (especially
  the natrural satellites).

  In addition, the header labels were obscure.

Details are given below.

Quantity #14 (sub-observer latitude and longitude)
============

* CURRENT headers (space and comma-delimited options):

   Case #1: NON-EARTH/MOON TARGET (IAU rotation model, 2 decimal places)

    SPACE: Ob-lon Ob-lat
    CSV  : Ob-lon, Ob-lat,

    Example of current output:
     SPACE: Ob-lon Ob-lat
            128.12 -17.12
     CSV  : Ob-lon, Ob-lat,
            128.12, -17.12,

   Case #2: EARTH/MOON TARGET (high-precision rot. model, 4 decimal places)

    SPACE: Obsrv-lon Obsrv-lat
    CSV  : Obsrv-lon, Obsrv-lat,

    Example of current output for Earth or Moon targets:
     SPACE: Obsrv-lon Obsrv-lat
            128.12345 -17.12345
     CSV  : Obsrv-lon, Obsrv-lat,
            128.12345, -17.12345,

* FUTURE headers (space and coma-delimited options):

  ALL rotatioal model cases

   SPACE: ObsSub-LON ObsSub-LAT
   CSV  : ObsSub-LON, ObsSub-LAT,

    Example of future output:
     SPACE: ObsSub-LON ObsSub-LAT
            128.123456 -17.123456
     CSV  : ObsSub-LON, ObsSub-LAT,
            128.123456, -17.123456,


Quantity #15 (sub-solar longitude and latitude)
============

* CURRENT headers (space and comma-delimited options):

   Case #1: NON-EARTH TARGET (IAU rotation model, 2 decimal places)
    SPACE: Sl-lon Sl-lat
    CSV  : Sl-lon, Sl-lat,

    Example of current output:
     SPACE: Sl-lon Sl-lat
            128.12 -17.12
     CSV  : Sl-lon, Sl-lat,
            128.12, -17.12,

   Case #2: EARTH TARGET (high-precision rot. model, 4 decimal places)
    SPACE: Solar-lon Solar-lat
    CSV  : Solar-lon, Solar-lat,

    Example of current output:
     SPACE: Solar-lon Solar-lat
            128.12345 -17.12345
     CSV  : Solar-lon, Solar-lat,
            128.12345, -17.12345,

* FUTURE headers (space and coma-delimited options):

  ALL rotatioal model cases
   SPACE: SunSub-LON SunSub-LAT
   CSV  : SunSub-LON, SunSub-LAT,

    Example of future output:
     SPACE: SunSub-LON SunSub-LAT
            128.123456 -17.123456
     CSV  : SunSub-LON, SunSub-LAT,
            128.123456, -17.123456,
```
Due to these changes, the following error is returned when attempting to parse the ephemeris retrieved by the jplhorizons routine:

```
.../astroquery/jplhorizons/core.py", line 1247, in _parse_horizons
    data[col].unit = column_defs[col][1]
KeyError: 'ObsSub-LON'
```

**Comment by bsipocz** [comment]
cc @mommermi

**Comment by mommermi** [comment]
... @bholler ... always causing trouble... :)

This should be fixed with #1650. Thanks for pointing out the error and the solution!

**Comment by HaralDev** [comment]
For clarity, this can be fixed by upgrading to the latest astroquery with:

> @Olivier-Lakoue - Please upgrade your astroquery with `pip install --pre -U astroquery`

[From #1650](https://github.com/astropy/astroquery/pull/1650#issuecomment-602894197)
